services/thermal_mgr/thermal_mgr.c

```diff
@@ -41,20 +42,72 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 }
 
+#define MAX_MS_WAIT 20
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+  if (!event)                 { return ERR_CODE_INVALID_ARG; }
+  if (!thermalMgrQueueHandle) { return ERR_CODE_INVALID_STATE; }
+
   /* Send an event to the thermal manager queue */
+  if (xQueueSend(thermalMgrQueueHandle, (void *)event,
+      portTICK_PERIOD_MS*MAX_MS_WAIT) == pdTRUE) {
+    // Successful send
+    return ERR_CODE_SUCCESS;
+  } else {
+    return ERR_CODE_QUEUE_FULL;
+  }
+}
 
-  return ERR_CODE_SUCCESS;
+error_code_t thermalMgrSendEventISR(thermal_mgr_event_t *event) {
+  /* Send an event to the thermal manager queue */
+  if (!event)                 { return ERR_CODE_INVALID_ARG; }
+  if (!thermalMgrQueueHandle) { return ERR_CODE_INVALID_STATE; }
+
+  if (xQueueSendFromISR(thermalMgrQueueHandle, (void *)event, NULL) == pdTRUE) {
+    return ERR_CODE_SUCCESS;
+  } else {
+    return ERR_CODE_QUEUE_FULL;
+  }
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  // Use new thermalMgr event for non-blocking.
+  thermal_mgr_event_t evt = {0};
+  evt.type = THERMAL_MGR_EVENT_OVER_TEMP_CMD;
+  thermalMgrSendEventISR(&evt);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t config = *(lm75bd_config_t *) pvParameters;
+  error_code_t errCode;
+
   while (1) {
-    
+    thermal_mgr_event_t tempEvt = {0};
+    if (
+        xQueueReceive(thermalMgrQueueHandle, (void *)&tempEvt,
+                      portTICK_PERIOD_MS*MAX_MS_WAIT) == pdFALSE) {
+      // Could not receive event
+      continue;
+    }
+
+    float temp = {0};
+    RETURN_IF_ERROR_CODE(readTempLM75BD(config.devAddr, &temp));
```

Comment:
This should just be a log if there's an error code followed by a continue since we should not be returning from FreeRTOS functions and especially since this is a void function and this would be returning an error code